### ISSUE TITLE
[RFC] [Inductor] decompose aten.log_normal_

### ISSUE THREAD

**Issue by min-jean-cho** [pytorch/pytorch#91275]
### 🚀 The feature, motivation and pitch

Currently aten.log_normal_ fallbacks to `FallbackKernel: torch.ops.aten.log_normal_.default`.

```python
import torch

def foo(x):
    return x.log_normal_()
    
opt_foo = torch.compile(foo)
x = torch.randn(100)
print(opt_foo(x))
```

`aten.log_normal_` is decomposable using `torch.randn` to something like:

```python
def log_normal(mean=1, std=2):
    return torch.exp(std * torch.randn(1) + mean)


x = []
for _ in range(10000):
    x.append(log_normal())

print(torch.mean(torch.tensor(x))) # verify E[X] is around e^(mean + 0.5 * std**2) ~= 20
print(torch.mean(torch.randn(10000).log_normal_())) # verify E[X] is around  e^(mean + 0.5 * std**2) ~= 20
```

### Correctness
```
X ~ Normal(mean, std**2), then
exp(X) ~ LogNormal(mean, std**2)
```

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @mlazos @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire @fritzo @neerajprad @alicanb @nikitaved

**Comment by min-jean-cho** [comment]
I am aware of the discussion regarding Prim Op --vs-- inductor-unique decomposed Aten Op, https://github.com/pytorch/torchdynamo/issues/327#issuecomment-1354409046, I've put the [Inductor] tag in the title as of now.

**Comment by min-jean-cho** [comment]
By the way, I think the documentation of LogNormal, https://pytorch.org/docs/stable/distributions.html#lognormal, is off:

```
X ~ Normal(loc, scale)
Y = exp(X) ~ LogNormal(loc, scale)
```
should be, 
```
X ~ Normal(loc, scale**2)
Y = exp(X) ~ LogNormal(loc, scale**2)
```

**Comment by min-jean-cho** [comment]
Hi @jgong5, @ashokei, please help let me know if any comments this.

**Comment by fritzo** [comment]
@min-jean-cho both `Normal` and `LogNormal` use `scale` rather than `variance = scale**2` as parameters.  This idiom follows `numpy.random`'s common parametrizations of (location,scale) families, and leads to cheaper computations than the other common idiom (mean,variance).

BTW what's the advantage of `.log_normal_()` over simply `.normal_().exp_()`?

**Comment by min-jean-cho** [comment]
> both Normal and LogNormal use scale rather than variance = scale**2 as parameters. This idiom follows numpy.random's common parametrizations of (location,scale) families

Thanks @fritzo, I think you are right - [location-scale family](https://en.wikipedia.org/wiki/Location%E2%80%93scale_family) are parameterized by `loc`, `scale`. I was off by thinking of:
```
N(µ, σ**2)
µ: location
σ: scale
```

**Comment by min-jean-cho** [comment]
> BTW what's the advantage of .log_normal_() over simply .normal_().exp_()?

Actually, `log_normal_` kernel samples from `normal_distribution` https://github.com/pytorch/pytorch/blob/release/1.13/aten/src/ATen/core/DistributionsHelper.h#L336 as `normal_` would, then transform it to log-normal https://github.com/pytorch/pytorch/blob/release/1.13/aten/src/ATen/core/TransformationHelper.h#L161 as `exp_` would.
